windows/ReactNativeCameraCPP/ReactCameraConstants.h

```diff
@@ -0,0 +1,140 @@
+
+    static std::map<std::wstring, std::wstring> GetBarcodeConstants() noexcept {
+        return std::map<std::wstring, std::wstring>
+        {
+            { L"UPC_E", L"32768" },
+            { L"CODE_39", L"4" },
+        };
+    }
+
+    static std::map<std::wstring, int> GetAutoFocusConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"off", CameraAutoFocusOff },
+            { L"on", CameraAutoFocusOn },
+        };
+    }
+
+    static std::map<std::wstring, int> GetWhiteBalanceConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"auto", CameraWhiteBalanceAuto },
+            { L"sunny", CameraWhiteBalanceSunny },
+            { L"cloudy", CameraWhiteBalanceCloudy },
+            { L"shadow", CameraWhiteBalanceShadow },
+            { L"incandescent", CameraWhiteBalanceIncandescent },
+            { L"fluorescent", CameraWhiteBalanceFluorescent },
+        };
+    }
+
+    static std::map<std::wstring, int> GetTypeConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"front", CameraTypeFront },
+            { L"back", CameraTypeBack },
+        };
+    }
+
+    static std::map<std::wstring, int> GetCaptureQualityConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"low", CameraCaptureQualityLow },
+            { L"high", CameraCaptureQualityHigh },
+            { L"720p", CameraCaptureQuality720p },
+            { L"1080p", CameraCaptureQuality1080p },
+        };
+    }
+
+    static std::map<std::wstring, int>   GetCaptureTargetConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"memory", CameraCaptureTargetMemory },
+            { L"disk", CameraCaptureTargetDisk },
+            { L"cameraRoll", CameraCaptureTargetCameraRoll },
+            { L"temp", CameraCaptureTargetTemp },
+        };
+    }
+
+    static std::map<std::wstring, int>   GetOrientationConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"auto", CameraOrientationAuto },
+            { L"landscapeLeft", CameraOrientationLandscapeLeft },
+            { L"landscapeRight", CameraOrientationLandscapeRight },
+            { L"portrait", CameraOrientationPortrait },
+            { L"portraitUpsideDown", CameraOrientationPortraitUpsideDown },
+        };
+    }
+
+    static std::map<std::wstring, int>   GetFlashModeConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"off", CameraFlashModeOff },
+            { L"on", CameraFlashModeOn },
+            { L"auto", CameraOrientationAuto },
+        };
+    }
+
+    static std::map<std::wstring, int>   GetTorchModeConstants() noexcept {
+        return std::map<std::wstring, int>
+        {
+            { L"off", CameraTorchModeOff },
+            { L"on", CameraTorchModeOn },
+            { L"auto", CameraTorchModeAuto },
+        };
+    }
+

```

Comment:
this is pretty wasteful and increases load/startup time
can we design this differently?

Reply:
This is to adhere to the current native module interface design, to return json values for contant definition. We can probably delay creating the std::maps when queried, but still needs time to create them at runtime.